x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -121,6 +121,20 @@ public void testAggsInsideGrouping() {
         );
     }
 
+    public void testGroupingInsideAggs() {
+        assertEquals(
+            "1:22: can only use grouping function [bucket(emp_no, 5.)] part of the BY clause",
+            error("from test| stats 3 + bucket(emp_no, 5.) by bucket(emp_no, 6.)")
```

Comment:
Please add a test that shows the following works:
`stats 3 + bucket(emp_no, 5) by bucket(emp_no, 5)` - namely repeating the grouping function in the by clauses, as an aggregation.
Also a test with bucket only inside the agg but not inside BY, plus another one without a BY clause.

Reply:
Added all these tests and a couple more. Updating the verifier was needed too.